code/datums/holocall.dm

```diff
@@ -0,0 +1,159 @@
+#define HOLOPAD_MAX_DIAL_TIME 200
+
+/mob/camera/aiEye/remote/holo/setLoc()
+	. = ..()
+	var/obj/machinery/hologram/holopad/H = origin
+	H.move_hologram(eye_user, loc)
+
+//this datum manages it's own references
+
+/datum/holocall
+	var/mob/living/user	//the one that called
+	var/obj/machinery/hologram/holopad/calling_holopad	//the one that sent the call
+	var/obj/machinery/hologram/holopad/connected_holopad	//the one that answered the call (may be null)
+	var/list/dialed_holopads	//all things called, will be cleared out to just connected_holopad once answered
+
+	var/mob/camera/aiEye/remote/holo/eye	//user's eye, once connected
+	var/obj/effect/overlay/holo_pad_hologram/hologram	//user's hologram, once connected
+
+	var/call_start_time
+
+//creates a holocall made by `caller` from `calling_pad` to `callees`
+/datum/holocall/New(mob/living/caller, obj/machinery/hologram/holopad/calling_pad, list/callees)
+	call_start_time = world.time
+	user = caller
+	calling_pad.outgoing_call = src
+	calling_holopad = calling_pad
+	dialed_holopads = list()
+
+	for(var/I in callees)
+		var/obj/machinery/hologram/holopad/H = I
+		if(!qdeleted(H) && !(H.stat & NOPOWER))
+			dialed_holopads += H
+			LAZYADD(H.holo_calls, src)
+
+	if(!dialed_holopads.len)
+		calling_holopad.visible_message("Connection failure.")
+		qdel(src)
+		return
+
+	log_debug("Holocall started")
```

Comment:
log_debug is only shown if admins have the "show debug logs" preference turned on, which it's off by default